### ISSUE TITLE
[Combat] TXMN with long names have level shown incorrectly

### ISSUE THREAD

**Issue by airon90** [Tuxemon/Tuxemon#239]
See the picture 

![schermata da 2016-08-26 14-20-17](https://cloud.githubusercontent.com/assets/139841/18005297/7f5aa1d8-6b99-11e6-91b8-0d0e8732c878.png)

**Comment by bitcraft** [comment]
This bug affects any TXMN with a long name.  I've noticed it with others as well.  Thanks for reporting this.

**Comment by Airon90** [comment]
I report everything I see, so that anyone could take up a bug and try to fix. I'd do it if only I were able to code seriously in Python >.<

**Comment by bitcraft** [comment]
Its great to have you reporting these bugs!  No worries.

**Comment by MirceaKitsune** [comment]
This appears to be fixed in the development branch, should this issue be closed? Here is a screenshot: I'm also fighting the Magmaturtle, but the level is no longer pushed all the way to the right... not sure what happens if the name is even longer than that thought.

![screenshot_20170513_010528](https://cloud.githubusercontent.com/assets/825418/26018658/db719a84-3778-11e7-9405-f87720aeed53.png)

**Comment by bitcraft** [comment]
There is still a limitation of long names, I think it is 12 characters.  I'd like to leave this open until a name length is agreed on.  Probably won't happen for a long time, as the monsters get developed.

**Comment by MirceaKitsune** [comment]
@bitcraft Fair enough. I can see two solutions to this limitation: Either just the name displayed in the battle menu is limited to 12 characters, either the fonts are squished horizontally to fit longer names (I've seen this done in other engines). Of course that's if the layout and font size are to be kept unmodified, bigger panels or smaller fonts could allow for two or more lines.

**Comment by bitcraft** [comment]
Its been a long time since looking at it.  I don't believe that it is solved currently because long names will be truncated.  A fix was implemented to prevent text from being drawn outside the border, but it will truncate long names.  A proper fix will be to horizontally compress/resize long names so they fit.  Leaving it open until that is implemented.

**Comment by Tello-** [comment]
I was curious about how this was handled in Pokémon and came across this little blurb on tvTropes:

> Pokémon was full of this, until later games expanded the character limit:
> Among Pokémon species, we have Feraligatr and Victreebel, as there's a limit of 10 letters for their names. Victreebel is particularly noticeable as its pre-evolution Weepinbell has both L's. The name limit was increased to 12 for Pokémon X and Y, but the abbreviated names remain.
> In the beta versions of the original games, Gyarados was called "Skulkraken" due to the same problem.
> Moves were often abbreviated this way thanks to a 12-character limit (for example, ThunderPunch and ExtremeSpeed). X and Y expanded the limit to 16, so many moves abbreviated this way were expanded up to take advantage of this (e.g. Thunder Punch and Extreme Speed).
> For human characters, the name limit was seven until Pokemon X and Y, where it was increased to eight.
> In the English version of Pokémon Stadium, Giovanni's name is spelled "Giovani" due to Trainer names being limited to seven characters.
> This is why the feather items introduced in Pokémon Black 2 and White 2 are called "Wings" in English and Paralysis Heal is called "Parlyz Heal" before being changed to "Paralyze Heal" in X and Y
> The notorious bootleg Pokémon Vietnamese Crystal is made even more nonsensical by retaining character limits obviously not designed for a Western language.
> "Pokémon" is sometimes shortened to "Pkmn" within the early games. To drive the point even further home, the P and K were often combined into one special character, while the M and N were combined into another.

Sorry for terrible formatting. I'm on mobile.

**Comment by Sanglorian** [comment]
Thanks for getting the conversation started again, Tello!

Looking at our current list of ~200 completed monsters and ~400 not completed ones, 99% have names 12 characters or fewer. 

Two have 13-character names: 

* Panthera fira
* Badderscratch

One has a 15-character name:  
* Quetzalrockilus

In the short term, I think renaming is easiest. Pantherafira, Baddrscratch, Qetzlrokilus. In the longer term though, it'd be good to allow longer lengths by squashing the text in. After all, we might want to add prefixes like "Mega " or "Fused " or something.

**Comment by Tello-** [comment]
I'm new around here, but I'll throw in my two cents. I think it would be better to set a solid limit on name length (as it seems you are leaning toward). Having names with more letters shrink in pixel size might not be too aesthetically pleasing.

Plus...limitations spark creativity. Cooler names may come out of the need to abbreviate in clever ways.

**Comment by Sanglorian** [comment]
Thanks for your two cents! I tend to agree.

The only caveat I have is that the Tuxemon project has collected monsters from a bunch of open source projects, not all of which will necessarily follow our rules. So in terms of what we include in the game, I think a 12 character limit is fine. But maybe on the wiki we should also acknowledge the longer names (perhaps under the "Alternative Names" section).

**Comment by Tello-** [comment]
That's a fair point about it being open source, and crowd sourced naming.

**Comment by Sanglorian** [comment]
I'd be happy to close this - of the 600+ monsters, the only ones with too-long names have been renamed. We could re-open if/when we want to accommodate longer names?

**Comment by Sanglorian** [comment]
Hi all, closing this one if there's no objections as every monster with a too-long name was renamed several years ago.
